File: mesmaeker-java-mockito/src/main/java/fr/xgouchet/mesmaeker/mockito/OngoingVoidClause.java
package fr.xgouchet.mesmaeker.mockito;

import java.util.function.Consumer;

/**
 * @author Xavier F. Gouchet
 */
public interface OngoingVoidClause<T> extends OngoingReverseClause<T> {

    OngoingVoidClause<T> when(Consumer<T> consumer);

}

File: mesmaeker-mockito/src/main/java/fr/xgouchet/mesmaeker/mockito/OngoingVoidClause.kt
package fr.xgouchet.mesmaeker.mockito

/**
 * @author Xavier F. Gouchet
 */
interface OngoingVoidClause<T> : OngoingReverseClause<T> {

    fun `when`(block: (T) -> Unit): OngoingVoidClause<T>

    fun whenever(block: (T) -> Unit): OngoingVoidClause<T>

}
